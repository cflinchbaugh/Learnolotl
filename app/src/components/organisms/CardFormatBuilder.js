import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

import DynamicFormElementFactory from './DynamicFormElementFactory';
import Button from 'buttons/Button';

const StyleWrapper = styled.div`
`

class CardFormatBuilder extends Component {
    constructor(props) {
        super(props);

        this.handleInputChange = this.handleInputChange.bind(this);
        this.saveFormat = this.saveFormat.bind(this);
        this.addInputField = this.addInputField.bind(this);

        this.state = {
            formElementData: {},
            formElements: [
                {
                    type: 'input',
                    data: {
                        id: 'revealOptionId',
                        label: 'Button Label',
                        placeholder: '',
                        value: ''
                    }
                }
            ]
        }
    }

    addInputField(e) {
        e.preventDefault();

        let newId = {
                type: 'input',
                data: {
                    id: 'revealOptionId',
                    placeholder: '',
                    label: 'Button Label',
                    value: ''
                }
            },
            updatedFormElements = [...this.state.formElements, newId];

        this.setState( (prevState, props) => {
            return {
                formElements: updatedFormElements
            };
        })
    }

    render() {
        let dynamicFormElementFactoryData = {
                handleInputChange: this.handleInputChange,
                formElementData: this.state.formElementData,
                addInputField: this.addInputField,
                formElements: this.state.formElements
                
            },
            saveFormatButtonData = {
                label: 'Save Format',
                onClickFunction: this.saveFormat
            }

        return (
            <StyleWrapper>
                    <DynamicFormElementFactory {...dynamicFormElementFactoryData}/>
                    <Button {...saveFormatButtonData}/>
            </StyleWrapper>

        )
    }

    saveFormat() {
        this.props.saveFormat(this.state.formElementData);
    }

    handleInputChange(formElementData) {
        let mergedData = {...this.state.formElementData, ...formElementData}

        this.setState( (prevState, props) => {
            return {
                formElementData: mergedData
            };
        });
    }

    

}

const mapStateToProps = state => ({
    buildResults: state.build.results
});

export default connect(mapStateToProps, { })(CardFormatBuilder);