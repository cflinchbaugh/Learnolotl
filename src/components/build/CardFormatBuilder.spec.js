import React from 'react';
import {shallow} from 'enzyme';
import CardFormatBuilder from './CardFormatBuilder';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('CardFormatBuilder', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();
        let cardFormatBuildData = {
            saveFormatButtonData: {
                type: 'next',
                onClickFunction: function(){}
            }
        }

        const tree = renderer.render(
            <CardFormatBuilder {...cardFormatBuildData}/>
        );

        expect(tree).toMatchSnapshot();
    });

});  