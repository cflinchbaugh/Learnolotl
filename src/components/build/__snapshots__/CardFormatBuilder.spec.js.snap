// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFormatBuilder renders and matches snapshot 1`] = `
<styled.div
  className="card-format-builder-wrapper clearfix"
>
  <h2>
    Card Format
  </h2>
  <p>
    The Modes built here will show as buttons on every card. When clicked, the hidden answer will appear.
  </p>
  <p>
    In the Japanese Sample, the Modes were "English", "Romaji", and "Hiragana".  For non-language learning, you may want to set your  modes to "Question" and "Answer."
  </p>
  <DynamicFormElementFactory />
  <NavigationArrow />
</styled.div>
`;
